File: Problems/Snakes and Ladders/Golang/board/board.go
package board

type Board struct {
	size int
}

func NewBoard(size int) *Board {
	return &Board{
		size: size,
	}
}

func (b *Board) GetSize() int {
	return b.size
}

func (b *Board) IsValidMove(playerPosition int, roll int) bool {
	if playerPosition == 0 && roll != 1 {
		return false
	}
	newPosition := playerPosition + roll
	return newPosition <= b.size
}

func (b *Board) CheckWinner(playerPosition int) bool {
	return playerPosition == b.size
}

File: Problems/Snakes and Ladders/Java/board/Board.java
package board;

public class Board {
    private int size;

    public Board(int size) {
        this.size = size;
    }

    public int getSize() {
        return size;
    }

    public boolean isValidMove(int playerPosition, int roll)  {
        if(playerPosition == 0 && roll != 1){
            return false;
        }

        int newPosition = playerPosition + roll;
        return newPosition <= size;
    }

    public boolean checkWinner(int playerPosition) {
        return playerPosition == size;
    }
}
